### ISSUE TITLE
Timing issues with core-toggle and its aria-label

### ISSUE THREAD

**Issue by wkillerud** [nrkno/core-components#628]
I wanted to run an issue I'm seeing in `@nrk/core-toggle` by y'all. We're using the React wrapper in [fremtind/jokul](https://github.com/fremtind/jokul) for our [Select](https://jokul.fremtind.no/komponenter/select) component. Because of a timing issue, `aria-label` is being set to include the value of the previously selected item instead of the currently selected.

The PR fremtind/jokul#2190 provides more context and a video from a debugging session.

What it boils down to is the source of the `aria-label` value and the timing at which `aria-label` is set. Our implementation ended up calling [this setter](https://github.com/nrkno/core-components/blob/d6420e204bbad6ada4bbf244897b825aad178f63/packages/core-toggle/core-toggle.js#L71) before our React-component had a chance to render with an updated value for `button.textContent`.

Is the below patch of interest upstream? 😄 If so I can open a PR.

```diff
  set value (data = false) {
    if (!this.button || !this.popup.length) return
    const button = this.button
    const popup = (button.getAttribute('aria-label') || `,${this.popup}`).split(',')[1]
    const label = data.textContent || data || '' // data can be Element, Object or String

    if (popup === this.popup) {
      const target = button.querySelector('span') || button // Use span to preserve embedded HTML and SVG
      button.value = data.value || label
      target[data.innerHTML ? 'innerHTML' : 'textContent'] = data.innerHTML || label
-     button.setAttribute('aria-label', `${button.textContent},${this.popup}`)
+     button.setAttribute('aria-label', `${label},${this.popup}`)
    }
  }
```

**Comment by skjalgepalg** [comment]
Hey, I'll look into this and get back to you with regards to a PR. Thanks for the heads-up!

**Comment by wkillerud** [comment]
Nice, thank you! And good to see a familiar face 😄 Let me know if I can help.

![](https://media.giphy.com/media/xT9IgG50Fb7Mi0prBC/giphy.gif?cid=ecf05e47o7murq5cgpd7iw5smt0akh7s3mv579oi60zjq7cw&rid=giphy.gif&ct=g)

**Comment by skjalgepalg** [comment]
@wkillerud reckon you could give it a try with @nrk/core-toggle@3.1.0?

Looks like the jsx -file published through npm for @nrk/core-toggle@3.0.8 sets aria-label prior to target being updated.
This was fixed in [0a3473](https://github.com/nrkno/core-components/commit/0a3473) but not released.
The correct order has been released as part of [core-toggle@3.1.0](https://unpkg.com/browse/@nrk/core-toggle@3.1.0/jsx.js)

Hopefully this should resolve your issue without the need to use `label` instead of `textContent`, but give me a shout if I've missed something

**Comment by wkillerud** [comment]
Can confirm `3.1.0` sets the correct `aria-label` on our end while still reading from `textContent` 🎉   @skjalgepalg
